### PR TITLE
Nested virtualization initial commit: part 1

### PR DESCRIPTION
**Pull request by ZideChen0**
The first part of nested virtualization.

### CODE REVIEW

**Comment by dongyaozu** [comment]
HV side is fine. Can @terryzouhao review the config tool part?

**Comment by wenlingz** [comment]
OK to verify

File: misc/hv_prebuild/vm_cfg_checks.c [hunk 1]
```diff
@@ -140,9 +140,21 @@ bool sanitize_vm_config(void)
 			}
 			break;
 		case SOS_VM:
-			if ((vm_config->severity != (uint8_t)SEVERITY_SOS) || ((vm_config->guest_flags & GUEST_FLAG_LAPIC_PASSTHROUGH) != 0U)) {
+			if (vm_config->severity != (uint8_t)SEVERITY_SOS) {
```
Comment (terryzouhao) [inline misc/hv_prebuild/vm_cfg_checks.c:143]: @ZideChen0 , yes we are switching to XML-based config tool, and feature/FLAG check could  be handled by XML schema, just like you did for SCHEDULER:  <xs:assert test="not (NVMX_ENABLED = 'y' and not (SCHEDULER = 'SCHED_NOOP'))"/> So could you move this sanity check into schema from vm_cfg_checks.c ?

File: misc/config_tools/data/tgl-rvp/hybrid.xml [hunk 2]
```diff
@@ -20,6 +20,7 @@
             <RDT_ENABLED>n</RDT_ENABLED>
             <CDP_ENABLED>n</CDP_ENABLED>
         </RDT>
+        <NVMX_ENABLED>n</NVMX_ENABLED>
```
Comment (terryzouhao) [inline misc/config_tools/data/tgl-rvp/hybrid.xml:23]: nested-virtualization is target enabled on tgl-hybird scenario first right, so should we set NVMX_enabled as 'y' by default?

**Review by terryzouhao — COMMENTED** [review]
Others are fine, only 2 comments for config tool:  prefer to do all sanity check in XML schema rather than in .c file, and do we need set 'NVMX_ENABLED' as 'y' for tgl-hybrid scenario.

File: misc/config_tools/data/tgl-rvp/hybrid.xml (hunk 2 above)
Comment (dongyaozu) [reply misc/config_tools/data/tgl-rvp/hybrid.xml:23]: Nested VMX feature is still an POC feature. So, it won't be turn on by default.

**Comment by ZideChen0** [comment]
Updated patch "misc: sanity check VM config for nested virtualization" to move the sanity check from vm_cfg_checks.c to schema.

File: misc/hv_prebuild/vm_cfg_checks.c (hunk 1 above)
Comment (terryzouhao) [reply misc/hv_prebuild/vm_cfg_checks.c]: Refined sanity check to XML schema, looks good to me now, thanks.

**Review by terryzouhao — APPROVED** [review]

**Pull request merged**